test/test_datasets_utils.py

```diff
@@ -177,6 +180,27 @@ def test_extract_archive_defer_to_decompress(self):
                     dict(from_path=file, to_path=filename, remove_finished=remove_finished),
                 )
 
+    def test_extract_rar(self):
+        rarfile = pytest.importorskip("rarfile")
+
+        def create_archive(root, content="this is the content"):
+            file = os.path.join(root, "dst.txt")
+            archive = os.path.join(root, "archive.rar")
+
+            subprocess.run(["rar", "a", archive, file], check=True)
```

Comment:
That would make our test suite dependent on this specific `rar` binary. Since `rarfile` supports more than just `rar`, is there way to detect the available backends with it and use whichever comes up?